src/py21cmfast/wrapper.py

```diff
@@ -2366,6 +2366,9 @@ def run_coeval(
         else:
             photon_nonconservation_data = None
 
+        if flag_options.USE_TS_FLUCT and user_params.USE_INTERPOLATION_TABLES:
+            lib.FreeTsInterpolationTables(flag_options())
```

Comment:
This is only for the `run_coeval` instance. You need to ensure this is also done for the `run_lightcone` as well.